src/mds/CInode.h

```diff
@@ -702,17 +725,36 @@ class CInode : public MDSCacheObject, public InodeStore {
   // client caps
   client_t loner_cap, want_loner_cap;
 
-  client_t get_loner() { return loner_cap; }
-  client_t get_wanted_loner() { return want_loner_cap; }
+  client_t get_loner() const { return loner_cap; }
+  client_t get_wanted_loner() const { return want_loner_cap; }
 
   // this is the loner state our locks should aim for
-  client_t get_target_loner() {
+  client_t get_target_loner() const {
     if (loner_cap == want_loner_cap)
       return loner_cap;
     else
       return -1;
   }
 
+  void state_clear(unsigned mask) {
+    MDSCacheObject::state_clear(mask);
+    if (!is_auth()) {
+      loner_cap = -1;
+    }
+  }
+  void state_set(unsigned mask) {
+    MDSCacheObject::state_set(mask);
+    if (!is_auth()) {
+      loner_cap = -1;
+    }
+  }
+  void state_reset(unsigned s) {
+    MDSCacheObject::state_reset(s);
+    if (!is_auth()) {
+      loner_cap = -1;
+    }
+  }
```

Comment:
Help me out here; why do we need to do this? Could it be broken out into a separate patch? :)